### PR TITLE
feat: Show missing values in red in `glimpse()`

### PR DESCRIPTION
**Pull request by ryanzomorrodi**
Changes to make glimpse show NAs in red., as brought up in #658.

### CODE REVIEW

File: R/glimpse.R [hunk 1]
```diff
@@ -144,7 +144,10 @@ format_glimpse.default <- function(x, ...) {
     dims_out <- paste0(dims, collapse = " x ")
     paste0("<", class(x)[[1]], "[", dims_out, "]>")
   } else {
-    format(x, trim = TRUE, justify = "none")
+    out <- format(x, trim = TRUE, justify = "none")
```
Comment (krlmlr) [inline R/glimpse.R:147]: This looks like an opportunity to remove the negation from the `if` condition.

File: R/glimpse.R [hunk 2]
```diff
@@ -169,14 +172,20 @@ format_glimpse.list <- function(x, ..., .inner = FALSE) {
...
 #' @export
 format_glimpse.factor <- function(x, ...) {
   if (any(grepl(",", levels(x), fixed = TRUE))) {
-    encodeString(as.character(x), quote = '"')
+    out <- encodeString(as.character(x), quote = '"')
   } else {
-    format(x, trim = TRUE, justify = "none")
+    out <- format(x, trim = TRUE, justify = "none")
   }
+  out[is.na(x)] <- crayon_red(NA)
```
Comment (krlmlr) [inline R/glimpse.R:188]: This looks like a pattern that we use in many places: assign a red `NA` string to a vector where another vector has `NA` values.

- Should this be a function?
- Should this be `crayon_red(NA)` or `crayon_red(out[is.na(x)])` instead?

**Review by krlmlr — COMMENTED** [review]
Thanks! Can you please add tests? There are some snapshot tests that also write the ANSI sequences to the snapshot file.

**Comment by ryanzomorrodi** [comment]
I added tests and a `style_na_if` function. I didn't use `crayon_red(out[is.na(x)])` because I was concerned about encoding the string adding quotes to NA, but it seems like it just transforms it to a character, so it seems like either would work.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**